frame/lending/src/mocks_offchain/wrapper.rs

```diff
@@ -0,0 +1,150 @@
+use serde::{Deserialize, Serialize};
+use sp_application_crypto::BoundToRuntimeAppPublic;
+use sp_core::{
+	crypto::{key_types, ByteArray, CryptoType, Dummy},
+	U256,
+};
+pub use sp_core::{sr25519, H256};
+use sp_runtime::{
+	codec::{Decode, Encode, MaxEncodedLen},
+	scale_info::TypeInfo,
+	testing::{TestSignature, UintAuthorityId},
+	traits::{IdentifyAccount, OpaqueKeys, Verify},
+	CryptoTypeId, KeyTypeId,
+};
+use std::{cell::RefCell, ops::Deref};
+
+#[derive(
+	Default,
+	PartialEq,
+	Eq,
+	Clone,
+	Encode,
+	Decode,
+	Debug,
+	Hash,
+	Serialize,
+	Deserialize,
+	PartialOrd,
+	Ord,
+	MaxEncodedLen,
+	TypeInfo,
+)]
+
+// The wrapper was added since AppCrypto is not implemented for UintAuthorityId
+pub struct UintAuthorityIdWrapper(pub UintAuthorityId);
+
+impl From<u64> for UintAuthorityIdWrapper {
+	fn from(id: u64) -> Self {
+		UintAuthorityIdWrapper(UintAuthorityId(id))
+	}
+}
+
+impl From<UintAuthorityIdWrapper> for u64 {
+	fn from(id: UintAuthorityIdWrapper) -> u64 {
+		id.0 .0
+	}
+}
+
+impl UintAuthorityIdWrapper {
+	/// Convert this authority ID into a public key.
+	pub fn to_public_key<T: ByteArray>(&self) -> T {
+		let bytes: [u8; 32] = U256::from(self.0 .0).into();
+		T::from_slice(&bytes).unwrap()
+	}
+}
+
+impl CryptoType for UintAuthorityIdWrapper {
+	type Pair = Dummy;
+}
+
+impl AsRef<[u8]> for UintAuthorityIdWrapper {
+	fn as_ref(&self) -> &[u8] {
+		unsafe {
+			let byte: *const _ = &(self.0 .0 as _);
+			std::slice::from_raw_parts(byte, std::mem::size_of::<u64>())
+		}
+	}
+}
+
+thread_local! {
+	static ALL_KEYS: RefCell<Vec<UintAuthorityIdWrapper>> = RefCell::new(vec![]);
+}
```

Comment:
Is the global state necessary? I'm not sure what the purpose is

Reply:
If it is necessary, perhaps add a comment explaining why?

Reply:
feels like this can be made state of struct?

Reply:
This is supposed to be a wrapper for https://docs.rs/sp-runtime/latest/sp_runtime/testing/struct.UintAuthorityId.html. I need it since UintAuthorityId does not implement frame_system::offchain::AppCrypto. I can  bring it to a more appropriate wrapper template form (for the moment it more looks like reimplementation). However, it will stay the same under the hood. Alternatively, I can completely reimplement this structure. What would be better solution?

Reply:
Implemented as a normal wrapper. If this approach is not suitable, please let me know